subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -10,28 +10,396 @@
+#include "ull_internal.h"
+#include "ull_adv_types.h"
+#include "ull_adv_internal.h"
+
+static struct ll_adv_iso ll_adv_iso[CONFIG_BT_CTLR_ADV_SET];
+static void *adv_iso_free;
+
+static MFIFO_DEFINE(iso_tx, sizeof(struct lll_tx),
+		    CONFIG_BT_CTLR_ISO_TX_BUFFERS);
+
+/* TODO: Share between BIS and CIS */
+static struct {
+	void *free;
+	uint8_t pool[CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE *
+			CONFIG_BT_CTLR_ISO_TX_BUFFERS];
+} mem_iso_tx;
+
+static struct ll_adv_iso *adv_iso_by_bis_handle_get(uint16_t bis_handle)
+{
+	struct ll_adv_iso *adv_iso;
+	uint8_t idx;
+
+	adv_iso =  &ll_adv_iso[0];
+
+	for (idx = 0U; idx < CONFIG_BT_CTLR_ADV_SET; idx++, adv_iso++) {
+		if (adv_iso->is_created &&
+		    (adv_iso->bis_handle == bis_handle)) {
+			return adv_iso;
+		}
+	}
+
+	return NULL;
+}
+
+void *ll_iso_tx_mem_acquire(void)
+{
+	return mem_acquire(&mem_iso_tx.free);
+}
+
+void ll_iso_tx_mem_release(void *tx)
+{
+	mem_release(tx, &mem_iso_tx.free);
+}
+
+static int init_reset(void)
+{
+	/* Initialize conn pool. */
+	mem_init(ll_adv_iso, sizeof(struct ll_adv_iso),
+		 sizeof(ll_adv_iso) / sizeof(struct ll_adv_iso), &adv_iso_free);
+
+	/* Initialize tx pool. */
+	mem_init(mem_iso_tx.pool, CONFIG_BT_CTLR_ISO_TX_BUFFER_SIZE,
+		 CONFIG_BT_CTLR_ISO_TX_BUFFERS, &mem_iso_tx.free);
+
+	return 0;
+}
+
+int ull_adv_iso_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int ull_adv_iso_reset(void)
+{
+	int err;
+
+	/* Re-initialize the Tx mfifo */
+	MFIFO_INIT(iso_tx);
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int ll_iso_tx_mem_enqueue(uint16_t handle, void *tx)
+{
+	struct lll_tx *lll_tx;
+	struct ll_adv_iso *adv_iso;
+	uint8_t idx;
+
+	adv_iso = adv_iso_by_bis_handle_get(handle);
+	if (!adv_iso) {
+		return -EINVAL;
+	}
+
+	idx = MFIFO_ENQUEUE_GET(iso_tx, (void **) &lll_tx);
+	if (!lll_tx) {
+		return -ENOBUFS;
+	}
+
+	lll_tx->handle = handle;
+	lll_tx->node = tx;
+
+	MFIFO_ENQUEUE(iso_tx, idx);
+
+	return 0;
+}
+
+uint8_t ll_adv_iso_by_hci_handle_get(uint8_t hci_handle, uint8_t *handle)
+{
+	struct ll_adv_iso *adv_iso;
+	uint8_t idx;
+
+	adv_iso =  &ll_adv_iso[0];
+
+	for (idx = 0U; idx < CONFIG_BT_CTLR_ADV_SET; idx++, adv_iso++) {
+		if (adv_iso->is_created &&
+		    (adv_iso->hci_handle == hci_handle)) {
+			*handle = idx;
+			return 0;
+		}
+	}
+
+	return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+}
+
+uint8_t ll_adv_iso_by_hci_handle_new(uint8_t hci_handle, uint8_t *handle)
+{
+	struct ll_adv_iso *adv_iso, *adv_iso_empty;
+	uint8_t idx;
+
+	adv_iso = &ll_adv_iso[0];
+	adv_iso_empty = NULL;
+
+	for (idx = 0U; idx < CONFIG_BT_CTLR_ADV_SET; idx++, adv_iso++) {
+		if (adv_iso->is_created) {
+			if (adv_iso->hci_handle == hci_handle) {
+				return BT_HCI_ERR_CMD_DISALLOWED;
+			}
+		} else if (!adv_iso_empty) {
+			adv_iso_empty = adv_iso;
+		}
+	}
+
+	if (adv_iso_empty) {
+		memset(adv_iso_empty, 0, sizeof(*adv_iso_empty));
+		adv_iso_empty->hci_handle = hci_handle;
+		*handle = idx;
```

Comment:
`idx` is always incremented to `CONFIG_BT_CTLR_ADV_SET`, so I don't think this does what it should do.